Multi-wallet support via EIP-6963 + wallet-aware UI/UX, deposit amount slider, and clearer BTC confirmation messaging

romanornr opened the pull request:
### Multi-wallet support via EIP-6963 + wallet-aware UI/UX, deposit amount slider, and clearer BTC confirmation messaging


- Multi-wallet support (Rabby Wallet, Coinbase Wallet, MetaMask) through EIP-6963 discovery
- In the future, other wallets can easily be supported 
- UI now shows the actual wallet the user connected with (including icon) and removes the need to ship static SVG assets per wallet.
- Deposit form with an amount slider, and amounts above spendable balance are auto-corrected down to the maximum allowed.
- Another issue happened when a user received their BTC from a faucet and immediately attempted to deposit it, resulting in a generic error message stating "No UTXOs found," which confused many users. But the issue was the required confirmations, the toast message was confusing. So I made it a bit more explicit, noting they just have to wait  


<img width="538" height="540" alt="image" src="https://github.com/user-attachments/assets/877860e5-1eb5-45b2-8e9f-60dffeadf54f" />


So Toast messaging about connected to Metamask now shows which specific wallet they're connected to 


Another feature is a slider for the deposit amount. 
Also, whenever a user fills in an amount that's too high, autocorrect the amount down to the maximum amount  
Here's how it looks 

<img width="717" height="613" alt="image" src="https://github.com/user-attachments/assets/d3faac59-69ca-47da-8b2e-e4a9e9dd568a" />


Note:
- Some variable names within the code still have to be changed from Metamask to a name that's more neutral 
- There's an edge case. E.g., a user can be connected to Metamask, disconnect, and decide to turn off Metamask extension while re-enabling his Rabbywallet. This change won't be picked up. I wasn't able to solve that yet, figured I might as well look at this minor edge case later. This edge case can be solved with the user just refreshing their browser, something they will probably do 'just in case' when turning on and off extensions

0xatomFusion reviewed (changes requested):
Looks great!!
Please, fix the errors when build the project `npm run build` and probably better to change the slider color from green to blue to keep the color scheme

romanornr commented:
Please don't deploy this yet 
Tried to make the balance slider thing reusable for both the deposit form and withdrawal form 

There's a problem when using the withdrawal form with the deposit slider
After moving the balance slider when withdrawing, you can imagine what happens 

It will try to keep fetching fee as the slider has been moved
Need to think of a way to solve that (in a nice way) 

<img width="1456" height="429" alt="image" src="https://github.com/user-attachments/assets/ee7c72c4-5139-4ef3-a5c0-87d1ecd33429" />

0xatomFusion commented:
> Please don't deploy this yet Tried to make the balance slider thing reusable for both the deposit form and withdrawal form
> 
> There's a problem when using the withdrawal form with the deposit slider After moving the balance slider when withdrawing, you can imagine what happens
> 
> It will try to keep fetching fee as the slider has been moved Need to think of a way to solve that (in a nice way)
> <img alt="image" width="1456" height="429" src="https://private-user-images.githubusercontent.com/6548898/490577047-ee7c72c4-5139-4ef3-a5c0-87d1ecd33429.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgxMTgwMTUsIm5iZiI6MTc1ODExNzcxNSwicGF0aCI6Ii82NTQ4ODk4LzQ5MDU3NzA0Ny1lZTdjNzJjNC01MTM5LTRlZjMtYTVjMC04N2QxZWNkMzM0MjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MTdUMTQwMTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjY5NzkwYjgwNTNkMGJkZTM3YjNmZDU5NjU2ZDk4OGZlMzY3YjIxMzY2Yjk3ZWNhYjYyZjY0NTVjODg3MTNmYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.l_k6mw0wCd1gmCT1_u4Vfr--zuDlA-4YrpjmLY7e33A">

Ohh, I see each time the balance change a new request is sent to API.
I suggest to sent the request to the API after 500ms of waiting and freeze the slide until the response. Or you can also try to cancel and http call from the FE to help reduce and not spam the backend.

romanornr commented:
A custom hook useDebounce is probably what we were looking for as a solution 
It's like those websites where you have a search form, but it only starts suggesting once you finish typing or have a break 

Which then, ie shows the suggested videos or the actress's name sliding and close matches 
Before the need to press to enter without spamming the backend for every keystroke neccessarily

romanornr commented:
Only thing left now is storing feeReserve in a config file

romanornr commented:
used a chrome extention to simulate a fee-estimate api but figured it's best have mocked fee-estimate api to test locally 
The L1Amount issue is now fixed (the debounce did some weird things with the input) 
Added a vitest for this (needs better testing, not sure) 

Also avoid an api call when input isn't valid yet

- withdraw still uses window.ethereum 
Which isn't a problem for chrome but is for Brave. I think chrome uses metamask for window.ethereum while brave browser uses their own wallet for window.ethereum, must be why brave didn't react to withdrawing while chrome did 

- Maybe add some point, it makes more sense to have a ui component where the user can 'chose' his wallet 
Add this moment I have implemented a priority about which wallet to pick first if available and then second etc 


Another thing I noticed
The withdraw form debounces the user input and calls store.fetchFeeEstimation(amount).
- If the user types quickly (e.g., 0.001 ->  0.0012 -> 0.00123), multiple requests are fired 
- Due to network latency or timing, an older request (for 0.001) can finish after a newer request (for 0.00123) and overwrite the store with an outdated fee  
--> which leads to side effects 

I think I can fix that with an abortcontroller to cancel prevous requests before starting a new one 
Have a request ID (not sure if ++requestId is a good idea, maybe that leads to problems with concurrent users using the bridge at the same time)
